default.json

```diff
@@ -61,10 +65,22 @@
       "extractVersion": "^kustomize\/(?<version>.*)$",
       "groupName": "kustomize"
     },
+    {
+      "matchManagers": ["gomod"],
+      "matchPackagePatterns": ["^k8s.io"],
+      "groupName": "gomod-k8sio-dependencies",
+      "allowedVersions": "<0.26.0"
```

Comment:
Does this reflect all the `k8s.io` packages?

![image](https://user-images.githubusercontent.com/5452977/225897000-597aa392-17c1-485b-a7fe-1eb664c0976c.png)

Reply:
We don't need to restrict versions on `sigs.k8s.io`, I can group `sigs.k8s.io` separately though if we want.

Reply:
The main concern is that it seems to be quite a wide grouping. Would this not skip `k8s.io/klog` upgrades for example?

Reply:
It is kind of a trade off because listing them all is also labor intensive and probably needs some updating, while `k8s.io/klog` is almost never a direct dependency (if even ever)? If you have an example for a direct dependency, or see more benefits in the first approach, I can look at including/excluding parts.

Reply:
Sure - please go ahead with the merge.
If we find good enough examples we can follow-up to plug them on a case by case basis.

Reply:
I updated it with a list of specific packages, I excluded `k8s.io/client-go` as that seems to be regularly replaced with our own fork at `rancher/client-go`.